catboost/jvm-packages/catboost4j-prediction/src/native/ai_catboost_CatBoostJNIImpl.cpp

```diff
@@ -92,8 +93,13 @@ static void HashCatFeatures(
         // NOTE: instead of C-style cast `dynamic_cast` should be used, but compiler complains that
         // `_jobject` is not a polymorphic type
         const auto jcatFeature = (jstring)jenv->GetObjectArrayElement(jcatFeatures, i);
+        CB_ENSURE(jcatFeature, "got null array element");
```

Comment:
This is a wrong way to check for `null`, you should do `jenv->IsSameObject(jcatFeature, NULL) == JNI_FALSE`.

Reply:
I was reading https://books.google.com/books?id=8M3F_sSSvWkC&lpg=PA66&ots=Qn07SiJ2fk&pg=PA66#v=onepage&q&f=false and it mentioned that you can compare by equality unless you have weak references. I think that only applies if you store weak references in your JNI class - in either case it wouldn't apply here since all the references are strong. I'm happy to change it if that's the style you prefer though.